ql/src/java/org/apache/hadoop/hive/ql/metadata/DummyPartition.java

```diff
@@ -83,4 +84,14 @@ public List<String> getValues() {
     return values;
   }
 
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof DummyPartition) {
+      DummyPartition o = (DummyPartition) obj;
+      return Objects.equals(name, o.getName()) && 
+          ((partSpec == null && o.partSpec == null) || 
```

Comment:
can you use `Objects.equals(partSpec, o.partSpec`)?

Reply:
Done